utils/releaser/src/utils.rs

```diff
@@ -69,3 +70,72 @@ pub fn root() -> PathBuf {
     }
     project_root
 }
+
+pub fn determine_new_version(bump_type: Option<String>) -> Option<String> {
+    let mut current_version: Vec<i32> = core_version()
+        .split('.')
+        .map(|f| f.parse().unwrap())
+        .collect();
+
+    let bump_type = bump_type.unwrap_or_else(|| "patch".to_string());
+    match bump_type.as_ref() {
+        "major" => current_version[0] += 1,
+        "minor" => current_version[1] += 1,
+        "patch" => current_version[2] += 1,
+        _ => panic!(
+            "{} is an undefined version bump. accepted values are: major, minor, patch",
+            bump_type
+        ),
+    }
+
+    let new_version = current_version
+        .iter()
+        .map(|f| f.to_string())
+        .collect::<Vec<String>>()
+        .join(".");
+
+    Some(new_version)
+}
+
+pub fn edit_cargo_version(cargo_path: &str, version: &str) {
+    let mut server = fs::read_to_string([cargo_path, "Cargo.toml"].join(""))
+        .unwrap()
+        .parse::<Document>()
+        .unwrap();
+
+    server["package"]["version"] = value(version);
+
+    fs::write(cargo_path, server.to_string()).unwrap();
+}
+
+pub fn bump_versions(bump_type: Option<String>) {
+    let new_version = determine_new_version(bump_type).unwrap_or_else(core_version);
+    let new_version = new_version.as_str();
+
+    let cargos_to_update = vec![
+        "/clients/admin/",
+        "/clients/cli/",
+        "/clients/egui/",
+        "/server/server/",
+        "/libs/core/",
+    ];
+    for cargo_path in cargos_to_update {
+        edit_cargo_version(cargo_path, new_version);
+    }
+
+    //apple
+    //TODO figure out what sbuild number and appropriate plist
+    Command::new("/usr/libexec/Plistbuddy").args([
+        "-c",
+        "Set CFBundleVersion $SBUILD_NUMBER",
+        "$PLIST",
+    ]);
+
+    Command::new("/usr/libexec/Plistbuddy").args([
+        "-c",
+        "Set CFBundleShortVersionString $BUNDLE_SHORT_VERSION",
+        "$PLIST",
+    ]);
+    //android
+    Command::new("./gradlew").args(["wrapper", "--gradle-version", &core_version()]);
```

Comment:
pretty sure this is wrong, i'm pretty sure this changes the wrapper gradle version, not our project's version.

You may have to explore: https://stackoverflow.com/questions/21329849/set-android-app-version-using-gradle

Reply:
thanks for pointing that out. There's no gradle command that you call through cli to bump versions. I used regex instead